packages/foam-vscode/src/features/preview-navigation.ts

```diff
@@ -151,6 +151,11 @@ export const markdownItWithRemoveLinkReferences = (
         if (refKey.includes(ALIAS_DIVIDER_CHAR)) {
           delete state.env.references[refKey];
         }
+
+        // if we try to include this note, we can remove the reference as well
```

Comment:
let's elaborate a bit more (possibly with a reference to this PR) the reasoning about this logic.
let's put ourselves in the shoes of other developers and future selves who are trying to understand what we wanted to do, and why, here